ciao_contrib/cda/data.py

```diff
@@ -347,32 +233,31 @@ def is_format(self, formats):
         the file is one of these formats."""
         return self.fileformat in formats
 
-    def get_filesize(self, ftp):
+    def get_filesize(self, headers):
         """Returns the file size in bytes.
 
-        The ftp object should be a FTP object connected to the
-        archive.
+        The hdr value is added to the request header to enable the
+        user-agent to be changed (or any other header).
         """
 
-        if self.filesize is None:
-            # the ftp.size command seems to be returning unreliable
-            # results, so manually parse the list output to grab the
-            # value, falling back to size if necessary
-            #
-            path = os.path.join(ftp.basedir, self.path)
-            v3("Finding size of: {}".format(path))
+        if self.filesize is not None:
+            return self.filesize
 
-            def hdlr(line):
-                v3("Looking for a filesize in " + line)
-                toks = line.split()
-                self.filesize = int(toks[4])
+        V3("Finding size of: {}".format(self.url))
 
-            try:
-                ftp.retrlines("LIST " + path, hdlr)
-            except Exception:
-                v3("Problem parsing LIST for filesize of " + path)
-                self.filesize = ftp.size(path)
+        req = urllib.request.Request(self.url, headers=headers)
+        try:
+            with urllib.request.urlopen(req) as rsp:
```

Comment:
do you need to add the ssl unverified context handler here?

Reply:
Ahhh. I hadn't tested this in classic CIAO on ubuntu, just conda CIAO. DId you see this as en error, or just noted it from code review?